R/geom-marquee-repel.R

```diff
@@ -0,0 +1,466 @@
+#' @rdname geom_text_repel
+#' @export
+geom_marquee_repel <- function(
+    mapping = NULL, data = NULL, stat = "identity", position = "identity",
+    ...,
+    box.padding = 0.25, point.padding = 1e-6,
+    min.segment.length = 0.5,
+    arrow = NULL,
+    force = 1,
+    force_pull = 1,
+    max.time = 0.5,
+    max.iter = 10000,
+    max.overlaps = getOption("ggrepel.max.overlaps", default = 10),
+    nudge_x = 0,
+    nudge_y = 0,
+    xlim = c(NA, NA),
+    ylim = c(NA, NA),
+    na.rm = FALSE,
+    show.legend = NA,
+    direction = c("both","y","x"),
+    seed = NA,
+    verbose = getOption("verbose", default = FALSE),
+    inherit.aes = TRUE
+) {
+  rlang::check_installed("marquee", "for rendering rich text.")
+  if (!missing(nudge_x) || !missing(nudge_y)) {
+    if (!missing(position)) {
+      stop("Specify either `position` or `nudge_x`/`nudge_y`", call. = FALSE)
+    }
+    position <- position_nudge_repel(nudge_x, nudge_y)
+  }
+  # Warn about limitations of the algorithm
+  if (verbose && any(abs(data$angle %% 90) > 5)) {
+    message(
+      "ggrepel: Repulsion works correctly only for rotation angles multiple of 90 degrees"
+    )
+  }
+  layer(
+    data = data,
+    mapping = mapping,
+    stat = stat,
+    geom = GeomMarqueeRepel,
+    position = position,
+    show.legend = show.legend,
+    inherit.aes = inherit.aes,
+    params = list(
+      box.padding   = to_unit(box.padding),
+      point.padding = to_unit(point.padding),
+      min.segment.length = to_unit(min.segment.length),
+      arrow = arrow,
+      na.rm = na.rm,
+      force = force,
+      force_pull = force_pull,
+      max.time = max.time,
+      max.iter = max.iter,
+      max.overlaps = max.overlaps,
+      nudge_x = nudge_x,
+      nudge_y = nudge_y,
+      xlim = xlim,
+      ylim = ylim,
+      direction = match.arg(direction),
+      seed = seed,
+      verbose = verbose,
+      ...
+    )
+  )
+}
+
+#' GeomMarqueeRepel
+#' @format NULL
+#' @usage NULL
+#' @keywords internal
+#' @export
+GeomMarqueeRepel <- ggproto(
+  "GeomMarqueeRepel", Geom,
+
+  required_aes = c("x", "y", "label"),
+
+  default_aes = aes(
+    colour = "black", fill = NA, size = 3.88, angle = 0,
+    hjust = 0.5, vjust = 0.5, alpha = NA, family = "", lineheight = 1.2,
+    # marquee specific
+    style = NULL, width = NA,
```

Comment:
It seems to me that `style` is supposed to be a `marquee::style()` object.

But after playing with `width` I cannot figure out what this is supposed to do. I tried numbers between 0 and 1, and larger numbers like 5 or 200, but I don't understand what is happening to the output. Could you clarify? I'm trying to write documentation in the Examples vignette so people can understand these arguments.

Reply:
Thanks for the clarification! I had some trouble reproducing examples last night, so that got me quite confused with how to use `width` and `style` properly. I had to run and re-run a lot of examples many times before they were showing up the way I expected. In retrospect, I think the way you wrote the code with aesthetics is the way to go. Sorry! My mistake.